backend/src/main/java/dev/projects/backend/controller/AdController.java

```diff
@@ -0,0 +1,90 @@
+package dev.projects.backend.controller;
+
+import dev.projects.backend.collection.Advertisement;
+import dev.projects.backend.dto.AdvertisementDTO;
+import dev.projects.backend.enums.BusinessCategory;
+import dev.projects.backend.enums.PaymentCategory;
+import dev.projects.backend.service.AdService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/ad")
+public class AdController {
+
+    @Autowired
+    private AdService adService;
```

Comment:
thats one way to use DI. but i recommend changing to constrctor injecting. 
change to : 
`private final AdService adService`
in combination with the @RequiredArgsConstructor Annotation

Reply:
@RequiredArgsConstructor with private final AdService adService; has been implemented